packages/vhd-lib/src/chain.js

```diff
@@ -1,28 +1,30 @@
 import { dirname, relative } from 'path'
 
-import { VhdFile } from './'
+import { openVhd } from './'
 import { DISK_TYPE_DIFFERENCING } from './_constants'
+import { Disposable } from 'promise-toolbox'
 
 export default async function chain(parentHandler, parentPath, childHandler, childPath, force = false) {
-  const parentVhd = new VhdFile(parentHandler, parentPath)
-  const childVhd = new VhdFile(childHandler, childPath)
+  await Disposable.use(
+    [openVhd(parentHandler, parentPath), openVhd(childHandler, childPath)],
+    async ([parentVhd, childVhd]) => {
+      await childVhd.readHeaderAndFooter()
+      const { header, footer } = childVhd
 
-  await childVhd.readHeaderAndFooter()
-  const { header, footer } = childVhd
+      if (footer.diskType !== DISK_TYPE_DIFFERENCING) {
+        if (!force) {
+          throw new Error('cannot chain disk of type ' + footer.diskType)
+        }
+        footer.diskType = DISK_TYPE_DIFFERENCING
+      }
+      await childVhd.readBlockAllocationTable()
```

Comment:
Why this change?

Reply:
writeFooter need to know the end of data in `VhdFile`, the end of data is computed from the block allocation table. 
Also `setUniqueParentLocator` can move block , and need block allocation table

Reply:
Sorry, my comment is not on the correct line, I meant: why did you remove `parentVhd.readHeaderAndFooter()`

Reply:
it's done in `openVhd`